Source/OnlineSubsystemEpic/Private/OnlinePresenceEpic.cpp

```diff
@@ -277,152 +192,90 @@ FOnlinePresenceEpic::FOnlinePresenceEpic(FOnlineSubsystemEpic const* InSubsystem
-			if (eosResult == EOS_EResult::EOS_Success)
-			{
-				// Set the raw status message
-				EOS_PresenceModification_SetRawRichTextOptions setRawMessageOptions = {
-					EOS_PRESENCE_SETRAWRICHTEXT_API_LATEST,
-					TCHAR_TO_UTF8(*Status.StatusStr)
-				};
-				eosResult = EOS_PresenceModification_SetRawRichText(modHandle, &setRawMessageOptions);
-				if (eosResult == EOS_EResult::EOS_Success)
-				{
-					// Set all additional presence properties.
-					int32 recordCount = Status.Properties.Num();
-					EOS_Presence_DataRecord* recordArr = new EOS_Presence_DataRecord[recordCount];
-
-					// Create the records
-					int32 currentIdx = 0;
-					for (auto prop : Status.Properties)
-					{
-						EOS_Presence_DataRecord dataRecord = {
-							EOS_PRESENCE_DATARECORD_API_LATEST,
-							TCHAR_TO_UTF8(*prop.Key),
-							TCHAR_TO_UTF8(*prop.Value.ToString())
-						};
-						recordArr[currentIdx] = dataRecord;
-						currentIdx += 1;
-					}
-
-					EOS_PresenceModification_SetDataOptions setDataOpts = {
-						EOS_PRESENCE_SETDATA_API_LATEST,
-						recordCount,
-						recordArr
-					};
-					eosResult = EOS_PresenceModification_SetData(modHandle, &setDataOpts);
-					if (eosResult == EOS_EResult::EOS_Success)
-					{
-						// Finally update the presence itself.
-						EOS_Presence_SetPresenceOptions setPresenceOptions = {
-							EOS_PRESENCE_SETPRESENCE_API_LATEST,
-							epicNetId.ToEpicAccountId(),
-							modHandle
-						};
-						FPresenceAdditionalData* additionalData = new FPresenceAdditionalData{
-							this,
-							epicNetId,
-							Delegate
-						};
-						EOS_Presence_SetPresence(this->presenceHandle, &setPresenceOptions, additionalData, &FOnlinePresenceEpic::EOS_SetPresenceComplete);
-					}
-					else
-					{
-						error = TEXT("[EOS SDK] Couldn't update additional data");
-					}
-				}
-				else
-				{
-					error = TEXT("[EOS SDK] Couldn't update presence text.");
-				}
-			}
-			else
-			{
-				error = TEXT("[EOS SDK] Couldn't update presence status.");
-			}
+			UE_LOG_ONLINE_PRESENCE(Warning, TEXT("%s: could not subscribe to presence updates."), *FString(__FUNCTION__));
+			return;
 		}
 		else
 		{
-			error = TEXT("[EOS SDK] Couldn't created presence modification handle");
+			/*
+			 * Add a subscription to a presence event when a query needs to be done.
+			 * NOTE: Only add this subscription if one hasn't done it already.
+			 */
+			PresenceNotifications.Add(epicUser.ToEpicAccountId(), PresenceNotificationHandle);
 		}
-	}
-	else
-	{
-		error = TEXT("User does not a valid EpicAccountId");
-	}
 
-	if (!error.IsEmpty())
-	{
-		UE_LOG_ONLINE_PRESENCE(Warning, TEXT("%s encounted an error. Message: %s"), *FString(__FUNCTION__), *error);
-	}
-}
+		//TODO - I don't know if this is right to just access 0 but I can't get the num anywhere else - maybe we loop through all local players?
+		TSharedPtr<FUniqueNetId const> localUser = subsystem->GetIdentityInterface()->GetUniquePlayerId(0);
```

Comment:
The identity interface has the `FPlatformUserId GetPlatformUserIdFromUniqueNetId(const FUniqueNetId&) const;` method you can use. Any `FPlatformUserId` is just a TypeDef'd `int32`

Reply:
Ok - problem with this though. I don't have access to the local user id.

Reply:
I just looked into the OSS code to see what they actually want with this method. It's bugged as it is missing a parameter, which tells which user id's to query presence for.

So going by the comments in Epics code and the method header, `User` is the aforementioned local user and the missing `UserIds` parameter would be the users to query presence for.

Since we're missing a parameter we have to assume that a call to this method intends to query _all_ available presence information.

Reply:
Yes, we don't have this. You want to be able to use this for _any_ user that you want to get information about. So you have to pass in the remote user as the parameter. A hack fix I can do is just loop through all the local players - similar to what is in the GetPlatformUserId function, and just query loop through valid ones using GetUniquePlayerId.